lwt-unix/conduit_lwt_unix.ml

```diff
@@ -28,14 +28,23 @@ let () =
   with Not_found -> ()
 
 type tls_lib = | OpenSSL | Native | No_tls [@@deriving sexp]
-let tls_library = ref No_tls
-let () =
-    tls_library := try
-        match Sys.getenv "CONDUIT_TLS" with
-        | "native" | "Native" | "NATIVE" -> Native
-        | _ -> OpenSSL
-      with Not_found -> OpenSSL
-    (* TODO build time selection *)
+let default_tls_library =
+  try
+    match String.lowercase_ascii (Sys.getenv "CONDUIT_TLS") with
+    | "native" -> Native
+    | "openssl" | "libressl" -> OpenSSL
+    | "none" | "notls" -> No_tls
+    | _ -> OpenSSL
+  with Not_found ->
+    if Conduit_lwt_unix_ssl.available then
```

Comment:
The previous behavior and all documentation I found indicated that OpenSSL would be selected first, then ocaml-tls. I'd be happy to have ocaml-tls selected first but I didn't want to potentially break backwards compatibility since conduit is already at 1.x.